pkgs/by-name/cl/clash-verge-rev/package.nix

```
@@ -1,66 +1,173 @@
 
-    runHook postInstall
-  '';
+      runHook postBuild
+    '';
 
-  postFixup = ''
-    rm -f $out/bin/verge-mihomo
-    ln -sf ${lib.getExe clash-meta} $out/bin/verge-mihomo
-  '';
+    installPhase = ''
+      runHook preInstall
 
-  passthru.updateScript = nix-update-script { };
+      cp -r dist $out
 
+      runHook postInstall
+    '';
+  };
   meta = {
     description = "Clash GUI based on tauri";
     homepage = "https://github.com/clash-verge-rev/clash-verge-rev";
     license = lib.licenses.gpl3Plus;
     mainProgram = "clash-verge";
-    maintainers = with lib.maintainers; [ Guanran928 ];
+    maintainers = with lib.maintainers; [
+      Guanran928
+      bot-wxt1221
+    ];
     platforms = [ "x86_64-linux" ];
-    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
   };
-})
+in
+rustPlatform.buildRustPackage {
+  inherit
+    pname
+    version
+    src
+    meta
+    ;
+
+  sourceRoot = "${src.name}/src-tauri";
+
+  cargoLock = {
+    lockFile = ./Cargo-tauri.lock;
+    outputHashes = {
+      "sysproxy-0.3.0" = "sha256-TEC51s/viqXUoEH9rJev8LdC2uHqefInNcarxeogePk=";
+    };
+  };
+  env = {
+    OPENSSL_NO_VENDOR = 1;
+  };
+  postPatch = ''
+    substituteInPlace ./tauri.conf.json \
+      --replace-fail '"distDir": "../dist",' '"distDir": "${webui}",' \
+      --replace-fail '"beforeBuildCommand": "pnpm run web:build"' '"beforeBuildCommand": ""'
+
+    mkdir -p ./sidecar
+    ln -s ${mihomo}/bin/mihomo sidecar/verge-mihomo-${rust.envVars.rustHostPlatform}
+    ln -s ${mihomo}/bin/mihomo sidecar/verge-mihomo-alpha-${rust.envVars.rustHostPlatform} # people should override mihomo to use alpha build. Because nixpkgs can't maintain alpha build.
+
+    mkdir -p ./resources
+    ln -s ${v2ray-geoip}/share/v2ray/geoip.dat resources/geoip.dat
+    ln -s ${v2ray-domain-list-community}/share/v2ray/geosite.dat resources/geosite.dat
+    ln -s ${dbip-country-lite.mmdb} resources/Country.mmdb
+
+  '';
+  nativeBuildInputs = [
+    wrapGAppsHook3
+    pkg-config
+    rustPlatform.cargoSetupHook
+    openssl
+    makeShellWrapper
+    webkitgtk
+  ];
+  buildInputs = [
+    openssl
+    webkitgtk
+  ];
+
+  postInstall = ''
+    install -DT icons/128x128@2x.png $out/share/icons/hicolor/128x128@2/apps/clash-verge.png
+    install -DT icons/128x128.png $out/share/icons/hicolor/128x128/apps/clash-verge.png
+    install -DT icons/32x32.png $out/share/icons/hicolor/32x32/apps/clash-verge.png
+    ln -s ${service}/bin/clash-verge-service $out/bin/clash-verge-service
+    mv $out/bin/clash-verge $out/bin/.clash-verge-wrapped
+    makeShellWrapper $out/bin/.clash-verge-wrapped $out/bin/clash-verge \
+      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ libayatana-appindicator ]}" \
+  '';
+
+  passthru.updateScript = nix-update-script { };
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = pname;
+      exec = "clash-verge %u";
+      icon = pname;
```

```suggestion
      name = "clash-verge-rev";
      exec = "clash-verge %u";
      icon = "clash-verge-rev";
```
both values depend on settings not controlled by pname